src/bin/stratisd.rs

```diff
@@ -85,20 +123,49 @@ fn run() -> StratisResult<()> {
                          revents: 0,
                          events: libc::POLLIN,
                      });
-            1
+            2
         }
-        None => 0,
+        None => 1,
     };
 
     // Don't timeout if eventable lets us be event-driven
-    let poll_timeout = if engine_fds_end_idx != 0 { -1 } else { 10000 };
+    let poll_timeout = if engine_fds_end_idx != 1 { -1 } else { 10000 };
 
     loop {
+        // Process any udev block events
+        if fds[0].revents != 0 {
+            loop {
+                match udev.receive_event() {
+                    Some(event) => {
+                        if let Some((devnode, device)) = handle_udev_add(&event) {
+                            let pool_uuid = engine.borrow_mut().block_evaluate(devnode, device);
+
+                            // We need to pretend that we aren't using the variable _pool_uuid so
+                            // that we can conditionally compile out the register_pool when dbus
+                            // is not enabled.
+                            if let Some(_pool_uuid) = pool_uuid {
+                                #[cfg(feature="dbus_enabled")]
+                                libstratis::dbus_api::register_pool(&mut dbus_conn,
+                                                                    &Rc::clone(&engine),
+                                                                    &mut dbus_context,
+                                                                    &mut tree,
+                                                                    _pool_uuid,
+                                                                    &base_object_path)?;
+                            }
+                        }
+                    }
+                    None => {
+                        break;
+                    }
+                };
+            }
+        }
+
         // Unconditionally call evented() if engine has no eventable.
         // This looks like a bad idea, but the only engine that has
         // no eventable is the sim engine, and for that engine, evented()
         // is essentially a no-op.
-        if engine_fds_end_idx == 0 {
+        if engine_fds_end_idx == 1 {
```

Comment:
You need to make more changes in this code because of the use of indexes to mean really important stuff.

```
for pfd in fds[...engine_fds_end_idx]
```

is wrong now. In fact, looks like there needs to more changes in loop, as carefully thought out ```expect``` string is wrong as well.

I'm wondering if it would have been a little less bad if engine had come before udev. But if you're going to stick with this order, probably a constant called ```udev_fds_end_index``` would help.